From 644e3deafa3047609a96a1c5ee682aefec157a73 Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jekoritz@microsoft.com>
Date: Wed, 17 Apr 2024 11:10:13 -0700
Subject: [PATCH] Keep our System.Text.Json version used in HostModel in sync
 with MSBuild.

This follows the same pattern as Roslyn, and ensures that the binding redirects in MSBuild kick in on .NET Framework.

This also happens to help unblock https://github.com/dotnet/installer/pull/19491 by working around the downgrade problem.

Backport: https://github.com/dotnet/runtime/pull/101199
---
 Directory.Build.targets                                      | 5 +++--
 eng/Versions.props                                           | 1 +
 .../Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj   | 4 +---
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Directory.Build.targets b/Directory.Build.targets
index f731eedc390..1161d409dec 100644
--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -86,9 +86,10 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <!-- when building from source we need to use the current version of MetadataLoadContext as the toolset version, but source-build imports
-         another props file which overrides the SystemReflectionMetadataLoadContextVersion from Version.props so we can't set it there -->
+    <!-- when building from source we need to use the current version of various packages as the toolset version, but source-build imports
+         another props file which overrides the versions from Version.props so we can't set it there -->
     <SystemReflectionMetadataLoadContextToolsetVersion Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(SystemReflectionMetadataLoadContextVersion)</SystemReflectionMetadataLoadContextToolsetVersion>
+    <SystemTextJsonToolsetVersion Condition="'$(DotNetBuildSourceOnly)' == 'true'">$(SystemTextJsonVersion)</SystemTextJsonToolsetVersion>
   </PropertyGroup>
 
   <Target Name="ValidateTargetOSLowercase"
diff --git a/eng/Versions.props b/eng/Versions.props
index 46847dbbdb2..b959f0ffd9c 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -135,5 +135,6 @@
     <!-- Keep toolset versions in sync with dotnet/msbuild and dotnet/sdk -->
     <SystemCollectionsImmutableToolsetVersion>8.0.0</SystemCollectionsImmutableToolsetVersion>
+    <SystemTextJsonToolsetVersion>8.0.0</SystemTextJsonToolsetVersion>
     <SystemReflectionMetadataToolsetVersion>8.0.0</SystemReflectionMetadataToolsetVersion>
     <SystemReflectionMetadataLoadContextToolsetVersion>8.0.0</SystemReflectionMetadataLoadContextToolsetVersion>
     <!-- Runtime-Assets dependencies -->
diff --git a/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj b/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj
index fec324d46cc..55a33eb7e24 100644
--- a/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj
+++ b/src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj
@@ -20,7 +20,7 @@
 
   <ItemGroup>
     <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataToolsetVersion)" />
-    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
+    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonToolsetVersion)" />
   </ItemGroup>
 
   <ItemGroup>
